### ISSUE TITLE
Printing Gradient Values inside SGD

### ISSUE THREAD

**Issue by felipefariax** [fchollet/keras#4183]
Hello!

I want to make some modifications into SGD code. I need to print the gradient values.

```
def get_updates(self, params, constraints, loss):
    grads = self.get_gradients(loss, params)

   ######### HOW TO PRINT/get_value GRADS???

    self.updates = []

    lr = self.lr
    if self.inital_decay > 0:
        lr *= (1. / (1. + self.decay * self.iterations))
        self.updates .append(K.update_add(self.iterations, 1))

    # momentum
    shapes = [K.get_variable_shape(p) for p in params]
    moments = [K.zeros(shape) for shape in shapes]
    self.weights = [self.iterations] + moments
    for p, g, m in zip(params, grads, moments):
        v = self.momentum * m - lr * g  # velocity
        self.updates.append(K.update(m, v))

        if self.nesterov:
            new_p = p + self.momentum * v - lr * g
        else:
            new_p = p + v

        # apply constraints
        if p in constraints:
            c = constraints[p]
            new_p = c(new_p)

        self.updates.append(K.update(p, new_p))
    return self.updates
```

How can I do this? I have tried .eval() with no success.

**Comment by kgrm** [comment]
The `get_updates()` method doesn't return the actual gradients during training, it only initialises the tensors and symbolic functions needed to compute them.

**Comment by felipefariax** [comment]
@kgrm
Do I have another way to print/store the values of the gradients during each update?
